src/argilla/utils/modeling.py

```diff
@@ -0,0 +1,241 @@
+            fields = list({key for record in records for key in record.fields.keys()})
+        # Extract all metrics for each field
+        field_metrics = {
+            field: self._extract_metrics_for_single_field(records=records, field=field) for field in fields
+        }
+        field_metrics = {field: metrics for field, metrics in field_metrics.items() if metrics is not None}
+        # If there is only one field, return the metrics for that field directly
+        if len(field_metrics) == 1:
+            return list(field_metrics.values())[0]
+        # If there are multiple fields, combine metrics for each field into a single dataframe
+        final_metrics = pd.concat(field_metrics, axis=1, keys=field_metrics.keys())
+        return final_metrics
+
+    def _cast_to_python_types(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Convert integer, boolean and floats columns in a dataframe
+        to Python native types.
+
+        Args:
+            df (pd.DataFrame): The text descriptives dataframe.
+
+        Returns:
+            pd.DataFrame: The text descriptives dataframe with integer and boolean columns cast to Python native types.
+        """
+        # Select columns by data type
+        int_cols = df.select_dtypes(include=["int64"]).columns
+        bool_cols = df.select_dtypes(include=["boolean"]).columns
+        float_cols = df.select_dtypes(include=["float64"]).columns
+        # Cast integer columns to Python's native int type
+        df[int_cols] = df[int_cols].astype(int)
+        # Cast boolean columns to Python's native str type
+        df[bool_cols] = df[bool_cols].astype(str)
+        # Cast float columns to Python's native float type and round to 2 decimal places
+        df[float_cols] = df[float_cols].astype(float).round(2)
+        return df
+
+    def _create_metadata_properties(self, df: pd.DataFrame) -> List:
+        """
+        Generate metadata properties based on dataframe columns and data types.
+
+        Args:
+            df (pd.DataFrame): The text descriptives dataframe.
+
+        Returns:
+            List: A list of metadata properties.
+        """
+        properties = []
+        for col, dtype in df.dtypes.items():
+            name = col
+            title = name.replace("_", " ").title()
+            if dtype in ["object", "bool"]:
+                prop = TermsMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators, values=["True", "False"])
+            elif dtype == "int64":
+                prop = IntegerMetadataProperty(
+                    name=name, title=title, visible_for_annotators=self.visible_for_annotators
+                )
+            elif dtype == "float64":
+                prop = FloatMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
+            else:
+                print(f"Unhandled data type for column {col}: {dtype}")
```

Comment:
Normally we expect to use a logger, warning or raised error for unintended usage like this

Reply:
Replaced the print statement with logging.